### ISSUE TITLE
Exclusion rules do not appear to work

### ISSUE THREAD

**Issue by rgov** [eddyxu/cpp-coveralls#138]
```
coveralls --build-root $(pwd) --gcov $(which llvm-cov) --gcov-options gcov --verbose --exclude-pattern ".*"
```

This should skip every file, because the regex matches all paths, but instead I see it generating a ton of gcov files.

**Comment by rgov** [comment]
Indeed, `run_gcov` does not ever touch the values in `args.regexp`.

**Comment by adam-stamand** [comment]
You can reference my explanation here #145 

tldr; --exclude-pattern isn't used for selecting what gets run with gcov, as you've discovered. It's used for filtering out files/directories from the final report. You have to use --exclude to exclude things from being run with gcov.
